alica_engine/include/engine/scheduler/Job.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include "engine/AlicaClock.h"
+
+#include <iostream>
+#include <chrono>
+#include <ctime>
+#include <vector>
+#include <memory>
+#include <functional>
+
+namespace alica {
+namespace scheduler
+{
+struct Job
+{
+    Job(std::function<void()> cb, std::vector<std::weak_ptr<Job>> prerequisites)
+        : cb(cb)
+        , prerequisites(prerequisites)
+        , cancelled(false)
+        , isRepeated(false)
+        , inExecution(false)
+    {
+
+    }
+
+    std::function<void()> cb;
```

Comment:
std::function is rather pricey.
Consider a cheaper alternatives (my best bet is hand-rolling a version with a fixed size a no heap allocation)